returnn/tf/layers/signal_processing.py

```diff
@@ -722,7 +783,8 @@ def get_out_data_from_opts(cls, name, sources, repetitions, n_out=None, **kwargs
     assert not input_data.sparse
     return Data(
       name="%s_output" % name,
-      shape=[input_data.get_placeholder_as_batch_major().shape[1].value, input_data.get_placeholder_as_batch_major().shape[2].value * repetitions],
+      shape=[input_data.get_placeholder_as_batch_major().shape[1].value,
+             input_data.get_placeholder_as_batch_major().shape[2].value * repetitions],
```

Comment:
This code doesn't make sense. Why do you get the tensor? `input_data` already has that information. It's just `input_data.shape[0]` and `input_data.shape[1]`, right?

Reply:
Probably yes. But it might not be batch major, right?

Reply:
`data.shape` does not include the batch-dim, so time-dim vs batch-dim doesn't matter.

Reply:
Okay, I just use `input_data.shape[0]` and `input_data.shape[1]` now.